### DISCUSSION TITLE
What's the difference between "--continue_path" and "--restore_path"?

### DISCUSSION THREAD

**Discussion by dostee-hemin** [coqui-ai/TTS#1966] [General Q&A] [answered]
According to the [documentation](https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html), --continue_path is used to "Continue a previous run", while --restore_path is used to "Fine-tune a model", but I don't get what the exact difference is between the two training processes. Don't they both just start training from an existing model with new datasets and configurations?

**Comment by freezerain** [comment]
I found this code in **trainer.py**
```
    continue_path: str = field(
        default="",
        metadata={
            "help": "Path to a training folder to continue training. Restore the model from the last checkpoint and continue training under the same folder."
        },
    )
    restore_path: str = field(
        default="",
        metadata={
            "help": "Path to a model checkpoit. Restore the model with the given checkpoint and start a new training."
        },
    )
```
While its not answering the question, it may be usefull for context

**Comment by Ca-ressemble-a-du-fake** [comment, marked as answer]
I think you can use both, you will end with a training that continues. I always use restore flag to continue a training from a given checkpoint and then it starts from this checkpoint but creates a new directory. I know that new checkpoints will be stored in this new directory.

If you use continue flag, you will provide a directory (not a checkpoint). Then it won't store the new checkpoints in a new file but in the directory you have provided. Thus it will continue training as usual and from an outer point of view you cannot notice training was interrupted.

With restore path, the checkpoint file will start from 60k steps for example whereas with continue path it will start at 10k steps.

**Reply by dostee-hemin** [reply]
Makes sense, thank you so much for the clarification.

**Reply by thorstenMueller** [reply]
Hi @dostee-hemin ,
just to "promote" a video from my Youtube channel 😆 this video might help.

https://youtu.be/O6KxJR95WpE

**Reply by dostee-hemin** [reply]
Thank you so much @thorstenMueller , the video perfectly explained the difference between the two.
